### PR TITLE
Add Kafka to Kafka Template

### PR DESCRIPTION
**Pull request by Surabhiad**

### CODE REVIEW

File: v2/kafka-to-kafka/src/test/java/com/google/cloud/teleport/v2/templates/KafkaToKafkaIT.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                .addParameter("inputTopic", inputTopicName)
+                .addParameter("outputTopic", outputTopicName)
+                .addParameter(
+                    "sinkBootstrapServer",
+                    kafkaResourceManager.getBootstrapServers().replace("PLAINTEXT://", "")));
+
+    LaunchInfo info = launchTemplate(options);
+    assertThatPipeline(info).isRunning();
+    KafkaProducer<String, String> kafkaProducer =
+        kafkaResourceManager.buildProducer(new StringSerializer(), new StringSerializer());
```
Comment (an2x) [inline v2/kafka-to-kafka/src/test/java/com/google/cloud/teleport/v2/templates/KafkaToKafkaIT.java:101]: I'd also add a test for binary messages, not only strings.

File: v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/ConsumerProperties.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ConsumerProperties {
+
+  private static final Logger LOGG = LoggerFactory.getLogger(ConsumerProperties.class);
+
+  public static List<String> accessSecretVersion(String projectId, KafkaToKafkaOptions options) {
```
Comment (an2x) [inline v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/ConsumerProperties.java:36]: This code seems to be duplicated between ConsumerProperties and ProducerProperties, consider extracting it into a separate (parent?) class or combining Consumer/Producer properties somehow.

File: v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/options/KafkaToKafkaOptions.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  void setSecretIdSinkPassword(String secretIdSinkPassword);
+
+  @TemplateParameter.Text(
+      order = 10,
+      optional = true,
+      description = "versionid from the secret manager to get username for source",
+      helpText = "version of the secret"
+  )
+
+  String getVersionIdSourceUsername();
```
Comment (an2x) [inline v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/options/KafkaToKafkaOptions.java:129]: As discussed offline, for Username and other params, we should only have a single "secretId" parameter.

**Review by an2x — CHANGES_REQUESTED** [review]

File: v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/Credentials.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import com.google.cloud.teleport.v2.utils.SecretManagerUtils;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class {@link Credentials} returns the SASL_PLAIN usernames and passwords
+ * for Kafka source and destinations.
+ */
+
+ final class Credentials{
```
Comment (an2x) [inline v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/Credentials.java:13]: As discussed offline, remove this and call SecretManagerUtils directly.

File: v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/ProducerProperties.java [hunk 5]
```diff
@@ -13,72 +13,40 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
+
```
Comment (an2x) [inline v2/kafka-to-kafka/src/main/java/com/google/cloud/teleport/v2/templates/ProducerProperties.java:16]: Please move this to the class javadoc and expand a little.

**Review by an2x — CHANGES_REQUESTED** [review]

**Review by an2x — APPROVED** [review]

**Comment by Abacn** [comment]
KafkaToKafkaIT.testKafkaToKafka: KafkaToKafkaIT introduced in this PR failing on main branch:

```
java.lang.IllegalArgumentException: Missing required value for [--migrationType, ""].
```

**Pull request merged**